src/main/org/codehaus/groovy/transform/stc/StaticTypeCheckingVisitor.java

```diff
@@ -171,7 +171,7 @@
 
     protected final ReturnAdder.ReturnStatementListener returnListener = new ReturnAdder.ReturnStatementListener() {
         public void returnStatementAdded(final ReturnStatement returnStatement) {
-            ClassNode returnType = checkReturnType(returnStatement);
+            checkReturnType(returnStatement);
```

Comment:
while true, that the variable is unused... for debugging purposes it helps a lot to have that information there